flux/src/flagger/canaries.tsx

```diff
@@ -0,0 +1,202 @@
+import { Table, SectionBox, Loader, Link } from '@kinvolk/headlamp-plugin/lib/components/common';
+import { Icon } from '@iconify/react';
+import CanaryStatus from './canarystatus';
+import { DeploymentProgress } from './deploymentprogress';
+import FlaggerAvailabilityCheck from './availabilitycheck';
+
+export default function Canaries() {
+    const [canary, error] = K8s.ResourceClasses.CustomResourceDefinition.useGet('canaries.flagger.app');
+    const canaryResourceClass = React.useMemo(() => {
+        return canary?.makeCRClass();
+    }, [canary]);
+
+    return (
+      <FlaggerAvailabilityCheck>
+       {canaryResourceClass ? <CanaryList canaryResourceClass={canaryResourceClass} /> : <Loader />}
+      </FlaggerAvailabilityCheck>
+    );
+}
+
+function CanaryList({ canaryResourceClass }) {
+    const [canaries] = canaryResourceClass.useList();
+    const [deployments] = K8s.ResourceClasses.Deployment.useList();
+    
+    // Calculate pod counts for canary and primary deployments
+    const canariesWithPodCounts = React.useMemo(() => {
+        if (!canaries || !deployments) return null;
+        return canaries.map(canary => {
+            const targetName = canary?.jsonData?.spec?.targetRef?.name;
+            const namespace = canary?.jsonData?.metadata?.namespace;
+            
+            // Find the original canary deployment
+            const canaryDeployment = deployments.find(
+                d => d.jsonData.metadata.name === targetName && d.jsonData.metadata.namespace === namespace
+            );
+            // Find the primary deployment (with -primary suffix)
+            const primaryDeployment = deployments.find(
+                d => d.jsonData.metadata.name === `${targetName}-primary` && d.jsonData.metadata.namespace === namespace
+            );
+            
+            const abTesting = canary?.jsonData?.spec?.analysis?.match || [];
+            
+            // Check if A/B testing is enabled
+            const hasAbTesting = abTesting.length > 0;
+            
+            // Format A/B testing headers or cookies
+            const abHeaders = abTesting
+                .filter(match => match.headers)
+                .map(match => Object.keys(match.headers)[0])
+                .join(', ');
+                
+            const abCookies = abTesting
+                .filter(match => match.cookies)
+                .map(match => Object.keys(match.cookies)[0])
+                .join(', ');
+
+
+            return {
+                ...canary.jsonData,
+                canaryPodCount: canaryDeployment?.status?.readyReplicas || 0,
+                primaryPodCount: primaryDeployment?.status?.readyReplicas || 0,
+                hasAbTesting,
+                abHeaders,
+                abCookies
+            };
+        });
+    }, [canaries, deployments]);
+    console.log("canariesWithPodCounts are ",canariesWithPodCounts);
+    return (
+        <SectionBox title="Canaries">
+            <Table
```

Comment:
Seems like we have 4 space iteration in this file but normally we do 2. Maybe the plugin has not been formatted.